rust-runtime/inlineable/src/client_request_compression.rs

```diff
@@ -76,10 +77,9 @@ impl Intercept for RequestCompressionInterceptor {
         "RequestCompressionInterceptor"
     }
 
-    fn read_before_serialization(
+    fn read_before_execution(
```

Comment:
**Question**: Why did this change? (not that it's wrong just curious)

Reply:
Ah good question. This change wasn't really necessary, but my thinking was
- `CompressionOptions` can be created at an early point in the sequence of interceptor execution, so we don't have to wait till a later interceptor method
-  Creating `CompressionOptions` doesn't require `runtime_components` so it can be done in an earlier interceptor method
- By creating CompressionOptions earlier, we allow for a larger gap between its creation and the modify_before_retry_loop, which means we can potentially run more interceptors in between in the future.

Reply:
Ah good question. This change wasn't really necessary, but my thinking was
- `CompressionOptions` can be created at an early point in the sequence of interceptor execution, so we don't have to wait till a later interceptor method
-  Creating `CompressionOptions` doesn't require `runtime_components` so it can be done in an earlier interceptor method
- By creating `CompressionOptions` earlier, we allow for a larger gap between its creation and the `modify_before_retry_loop`, which means we can potentially run more interceptors in between in the future